clients/java/src/main/java/marquez/client/HttpBackend.java

```diff
@@ -36,6 +36,11 @@ public URL getBaseUrl() {
 
   private URL url(String path) {
     try {
+      if (baseUrl.toString().endsWith("/") && path.startsWith("/")) {
```

Comment:
The caller passes in the `baseUrl`, so I agree stripping the `/` character is a valid check and should be removed. But, we control the `path` and also the base path (=`/api/v1`). So, I'm not sure changing the paths in the client makes sense given that they are internal and never exposed to the caller. Mind reverting the path changes in this PR, but still keep the stripping of the trailing `/` in the `baseUrl`?

Reply:
I'd also add the logic to strip the trailing `/` when a [new `MarquezUrl`](https://github.com/MarquezProject/marquez/blob/main/clients/java/src/main/java/marquez/client/MarquezUrl.java#L45) object is constructed rather than on each call to `put()`

Reply:
Done in Utils.toUrl() which is MarquezUrl's package-private API.